src/qtt/algorithms/allxy.py

```diff
@@ -49,13 +49,13 @@ def allxy_model(indices: Union[float, np.ndarray], offset0: float, slope0: float
     return v1 + v2 + v3
 
 
-def _estimate_allxy_parameters(allxy_data: np.ndarray) -> List[float]:
+def _estimate_allxy_parameters(allxy_data: np.ndarray) -> List[Any]:
     """ Return estimate of allxy model parameters """
     p = [np.mean(allxy_data[0:5]), 0, np.mean(allxy_data[5:17]), 0, np.mean(allxy_data[17:]), 0]
     return p
 
 
-def _default_measurement_array(dataset: DataSet) -> DataArray:
+def _default_measurement_array(dataset: DataSet) -> np.ndarray:
```

Comment:
The `_default_measurement_array` does return a `DataArray`. Annotating it as `np.ndarray` is not entirely correct, but ok since the DataArray converts to `np.ndarray` automatically.

The way to do it would be to use the `ArrayLike` type (see https://numpy.org/devdocs/reference/typing.html). This requires a more recent numpy though, so suggest to leave the annotation as it is.